### PR TITLE
Inversion of Control dependency injection and form validation

### PR DESCRIPTION
**Pull request by annemca**
Created service that calls fake api (npm server-json). Posts form data and redirects to a summary of form input. Npm inversify for inversion of control dependency injection into formExampleController. Mocked service in tests using npm ts-mockito. Form validation using npm class-validation.

### CODE REVIEW

**Comment by shaunganley** [comment]
checked out source and ran tests, got the following error, looks like tslint package isn't saved in package.json

**Comment by shaunganley** [comment]
bug: select preferred contact option, don't enter value. Error displayed `Error: Post rejected`

**Comment by shaunganley** [comment]
finished commenting, just a few small comments, great work

File: package.json [hunk 1]
```diff
@@ -7,10 +7,12 @@
     "clean": "rimraf dist",
     "lint": "tslint --project tslint.json",
     "start": "npm run build && node ./bin/www",
+    "fakeApi": "json-server -p 4000 db.json",
     "test": "npm run lint && mocha tests/unit/**/*.ts --require ts-node/register",
     "generate-assets": "gulp generate-assets"
   },
   "dependencies": {
+    "axios": "^0.18.0",
```
[the remark this answers is not in the document]
Comment (annemca) [reply package.json]: Axios is very popular and very easy to implement, it supports promises and async/await and automatically transforms json data. What is chosen for http will depend on balancing the requirements of an individual project though. Here is some (quick) background reading that gives axios some context:

https://flaviocopes.com/axios/
https://medium.com/@thejasonfile/fetch-vs-axios-js-for-making-http-requests-2b261cdd3af5
https://www.valentinog.com/blog/http-requests-node-js-async-await/
https://www.twilio.com/blog/2017/08/http-requests-in-node-js.html

**Pull request merged**